apps/chatbot/src/modules/presidio.py

```diff
@@ -0,0 +1,215 @@
+    "US_ITIN",
+    "US_PASSPORT",
+    "US_SSN",
+    "UK_NHS"
+]
+
+ALLOW_LIST = [
+    "Discovery", "discovery", "pagoPA", "PagoPA", "pagopa"
+]
+
+
+class EntityTypeCountAnonymizer(Operator):
+    """
+    Anonymizer which replaces the entity value
+    with an type counter per entity.
+    """
+
+    REPLACING_FORMAT = "<{entity_type}_{index}>"
+
+    def operate(self, text: str, params: Dict[str, Any]) -> str:
+        """Anonymize the input text."""
+        entity_type: str = params["entity_type"]
+        entity_mapping: Dict[str, Dict] = params["entity_mapping"]
+        deanonymize_mapping: Dict[str, str] = params["deanonymize_mapping"]
+
+        entity_mapping_for_type = entity_mapping.get(entity_type)
+        if not entity_mapping_for_type:
+            entity_mapping_for_type = entity_mapping[entity_type] = {}
+
+        if text in entity_mapping_for_type:
+            return entity_mapping_for_type[text]
+
+        new_text = self.REPLACING_FORMAT.format(
+            entity_type=entity_type, index=len(entity_mapping_for_type) + 1
+        )
+        entity_mapping[entity_type][text] = new_text
+        deanonymize_mapping[new_text] = text
+        return new_text
+
+    def validate(self, params: Dict[str, Any]) -> None:
+        """Validate operator parameters."""
+        if "entity_mapping" not in params:
+            raise ValueError("An input Dict called `entity_mapping` is required.")
+        if "entity_type" not in params:
+            raise ValueError("An entity_type param is required.")
+        if "deanonymize_mapping" not in params:
+            raise ValueError("A deanonymize_mapping param is required.")
+
+    def operator_name(self) -> str:
+        return self.__class__.__name__
+
+    def operator_type(self) -> OperatorType:
+        return OperatorType.Anonymize
+
+
+class PresidioPII():
+    """Uses a presidio to analyse PIIs.
+    """
+
+    def __init__(
+            self,
+            config: Union[Path, str] | dict,
+            entity_mapping: Dict[str, Dict] = {},
+            mapping: Dict[str, str] = {},
+            entities: List[str] | None = None,
+            analyzer_threshold: float = 0.4
+        ):
+        self.config = config
+        self.entity_mapping = entity_mapping
+        self.mapping = mapping
+        self.entities = entities if entities else ENTITIES
+        self.analyzer_threshold = analyzer_threshold
+
+        if isinstance(self.config, (Path, str)):
+            self.provider = NlpEngineProvider(conf_file=self.config)
+        elif isinstance(self.config, dict):
+            self.provider = NlpEngineProvider(nlp_configuration=self.config)
+        else:
+            raise ValueError("Error! config should be a path or a dictionary.")
+        nlp_engine = self.provider.create_engine()
+        self.nlp_engine = nlp_engine
+        self.analyzer = AnalyzerEngine(
+            nlp_engine = self.nlp_engine,
+            supported_languages = ["it", "en", "es", "fr", "de"],
```

Comment:
Is it actually necessary to support other languages than italian?

Reply:
I do not know... There isn't anymore a block language, so a user can write using any language. Presidio masks PII entites only if the language is one of its inputs (see `detect_pii` method). So, I assumed that a normal PagoPA user could speak one of the main european languages.